src/client/client.cc

```diff
@@ -218,7 +245,25 @@ namespace Pistache::Http::Experimental
                 , connection(connection)
                 , addr_len(_addr_len)
             {
-                std::memcpy(&addr, _addr, addr_len);
+                // Note: Sanitizer may object to invocation of memcpy with null
+                // _addr, even if addr_len is zero
+                if (_addr_len)
+                {
+                    // Note - the "cast" here avoids a warning of mismatched
+                    // signedness in Windows
+                    if (addr_len > (static_cast<socklen_t>(sizeof(addr))))
```

Comment:
Here, with `sizeof(addr)`, you're taking the size of the pointer. Is this intentional?